src/api/app/models/bs_request.rb

```diff
@@ -608,6 +608,7 @@ def assignreview(opts = {})
 
   def change_review_state(new_review_state, opts = {})
     with_lock do
+    Review.transaction do
```

Comment:
yes, I avoided to indent the entire block because it makes review and blame way harder (and just steal commit line credentials;)